src/init/setup/SetupHandler.ts

```diff
@@ -0,0 +1,75 @@
+import { BasicRepresentation } from '../../http/representation/BasicRepresentation';
+import type { Representation } from '../../http/representation/Representation';
+import { BaseInteractionHandler } from '../../identity/interaction/BaseInteractionHandler';
+import type { RegistrationManager,
+  RegistrationParams } from '../../identity/interaction/email-password/util/RegistrationManager';
+import type { InteractionHandlerInput } from '../../identity/interaction/InteractionHandler';
+import { getLoggerFor } from '../../logging/LogUtil';
+import { APPLICATION_JSON } from '../../util/ContentTypes';
+import { NotImplementedHttpError } from '../../util/errors/NotImplementedHttpError';
+import { readJsonStream } from '../../util/StreamUtil';
+import type { Initializer } from '../Initializer';
+
+export interface SetupHandlerArgs {
+  /**
+   * Used for registering a pod during setup.
+   */
+  registrationManager?: RegistrationManager;
+  /**
+   * Initializer to call in case no registration procedure needs to happen.
+   * This Initializer should make sure the necessary resources are there so the server can work correctly.
+   */
+  initializer?: Initializer;
+}
+
+/**
+ * On POST requests, runs an initializer and/or performs a registration step, both optional.
+ */
+export class SetupHandler extends BaseInteractionHandler {
+  protected readonly logger = getLoggerFor(this);
+
+  private readonly registrationManager?: RegistrationManager;
+  private readonly initializer?: Initializer;
+
+  public constructor(args: SetupHandlerArgs) {
+    super({});
+    this.registrationManager = args.registrationManager;
+    this.initializer = args.initializer;
+  }
+
+  protected async handlePost({ operation }: InteractionHandlerInput): Promise<Representation> {
```

Comment:
Maybe split this one in 3 methods? Or have comment blocks explaining the steps? I wasn't sure what was coming.